### ISSUE TITLE
Show service details / runsheets in the public interface

### ISSUE THREAD

**Issue by jefft** [tbar0970/jethro-pmm#1003]
Jethro lets one plan details of what will happen in a service:

![image](https://github.com/tbar0970/jethro-pmm/assets/205995/8fead20e-1c22-4b4b-9a61-cef71d00a182)

but then these service runsheets are only visible to staff members with a full account. The majority of people the runsheet is meant to help cannot see it. AV/Computer people cannot see the order of events. Music people cannot see where their songs fit into the order (or songs at all, unless the 'All Songs' Service Field' is enabled in the roster).

Jethro can make rosters public, so why not also runsheets? In services, rosters say "who" and runsheets say "what" - they should both be visible together.  If the public interface is enabled, people should be able to browse upcoming services:

![image](https://github.com/tbar0970/jethro-pmm/assets/205995/04e1111d-a860-43c8-bdec-ced623fc508c)

and also have a 'Runsheet link' Service Field that can be added to roster views.

**Comment by simon-KAC** [comment]
Yes, a link from the public roster to a view only version or even the printable version of the Run Sheet would be great. The person on prayers could see if they are praying before or after the sermon and plan accordingly without waiting on the service leader to email out the Run Sheet.

**Comment by tbar0970** [comment]
Related thought I had previousy:  https://github.com/tbar0970/jethro-pmm/issues/659#issuecomment-1805258428

I agree it'd be good to streamline the process of people seeing the run sheet.

To my mind, run sheets are a *little* bit private and I wouldn't really want them world-visible. It's not that they have super-secret info, but they're kinda "in house business".  

Also, run sheets are often built up slowly - they might contain doodlings or vague plans to begin with, until they are published in final form.  So I wouldn't want over-eager people to read the run sheet early in the process and assume that's what the final version will be like.

**Comment by tbar0970** [comment]
> Yes, a link from the public roster to a view only version or even the printable version of the Run Sheet would be great. The person on prayers could see if they are praying before or after the sermon and plan accordingly without waiting on the service leader to email out the Run Sheet.

Good example! Do your service leaders only send out the run sheets quite late in the piece?

**Comment by jefft** [comment]
> To my mind, run sheets are a little bit private and I wouldn't really want them world-visible. It's not that they have super-secret info, but they're kinda "in house business".

Perhaps "accessible by the public if you know the link" is the correct solution. In your #659 example of "myjethro.com/public/?call=runsheet&id=abc213fgh890", just make `id` a uuid stored in the database identifying the runsheet.

Then have a 'public runsheet URL' button on the runsheet page, which can be pasted into emails. Also a 'Runsheet link' Service Field.

> Also, run sheets are often built up slowly - they might contain doodlings or vague plans to begin with, until they are published in final form. So I wouldn't want over-eager people to read the run sheet early in the process and assume that's what the final version will be like.

In our church, PDF runsheets go out on Wednesday, but come Sunday, the musicians have changed the songs, new notices have been added, and Jimmy and Sally swapped weeks serving. All up the PDF from Wednesday is pretty useless, and there is nothing else available because Jethro keeps runsheets so close to its chest. Giving people a link to the live runsheet as it evolves would be a great improvement.

**Comment by tbar0970** [comment]
> Perhaps "accessible by the public if you know the link" is the correct solution. In your https://github.com/tbar0970/jethro-pmm/issues/659 example of "myjethro.com/public/?call=runsheet&id=abc213fgh890", just make id a uuid stored in the database identifying the runsheet.

Yep.  If the user had to click a button to generate this, it would signify "this run sheet is ready for publication"

> Giving people a link to the live runsheet as it evolves would be a great improvement.

Yep
